test/suite/test_prepare22.py

```diff
@@ -0,0 +1,134 @@
+             value_c = 99
+        else:
+             value_a = "a"
+             value_b = "b"
+             value_c = "c"
+
+        # Pin oldest timestamp to 1
+        self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(1))
+
+        # Do the first update
+        cursor = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        cursor[1] = value_a
+        self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(10))
+
+        # Do the second update
+        self.session.begin_transaction()
+        cursor[1] = value_b
+        self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(20))
+
+        if self.delete:
+            self.session.begin_transaction()
+            cursor.set_key(1)
+            cursor.remove()
+            self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(30))
+
+        # Do a prepared update
+        self.session.begin_transaction()
+        cursor[1] = value_c
+        self.session.prepare_transaction('prepare_timestamp=' + self.timestamp_str(40))
+
+        # Evict the page
+        session2 = self.conn.open_session()
+        evict_cursor = session2.open_cursor(uri, None, 'debug=(release_evict)')
+        session2.begin_transaction('ignore_prepare=true,read_timestamp=' + self.timestamp_str(20))
+        self.assertEquals(evict_cursor[1], value_b)
+        evict_cursor.reset()
+        evict_cursor.close()
+        session2.rollback_transaction()
+
+        # Ensure the history store is checkpointed
+        session2.checkpoint()
+
+        # Rollback the prepared transaction
+        self.session.rollback_transaction()
+
+        # Set stable timestamp to 30
+        self.conn.set_timestamp('stable_timestamp=' + self.timestamp_str(30))
+
+        # Call rollback to stable
+        self.conn.rollback_to_stable()
+
+        # Evict the page again
+        evict_cursor = session2.open_cursor(uri, None, 'debug=(release_evict)')
+        session2.begin_transaction('read_timestamp=' + self.timestamp_str(20))
+        self.assertEquals(evict_cursor[1], value_b)
+        evict_cursor.reset()
+        evict_cursor.close()
+        session2.rollback_transaction()
+
+        # Verify we can still read back value a
+        self.session.begin_transaction('read_timestamp=' + self.timestamp_str(10))
+        self.assertEquals(cursor[1], value_a)
+        self.session.rollback_transaction()
+
+        # Verify we can still read back value b
+        self.session.begin_transaction('read_timestamp=' + self.timestamp_str(20))
+        self.assertEquals(cursor[1], value_b)
+        self.session.rollback_transaction()
+
+        # Verify we can still read back the deletion
+        if self.delete and self.value_format != '8t':
+            self.session.begin_transaction('read_timestamp=' + self.timestamp_str(30))
+            cursor.set_key(1)
+            self.assertEquals(cursor.search(), wiredtiger.WT_NOTFOUND)
```

Comment:
The following check can be added for fixed length column store also by checking the value as `0`. Check for RTS tests `check` function for reference.

Reply:
[2e4f53e](https://github.com/wiredtiger/wiredtiger/pull/8131/commits/2e4f53ea01ef1e455570272ac7e6f65b29a82d63)